### PR TITLE
black + isort everything

### PR DESCRIPTION
**Pull request by mr-c**
## Changelog Entry
To be copied to the [draft changelog](https://github.com/DataBiosphere/toil/wiki/Draft-Changelog) by merger:

 * [Misc] All source code is now formatted by `black`

## Reviewer Checklist

 * [ ] Make sure it is coming from `issues/XXXX-fix-the-thing` in the Toil repo, or from an external repo.
    * [ ] If it is coming from an external repo, make sure to pull it in for CI with:
        ```
        contrib/admin/test-pr otheruser theirbranchname issues/XXXX-fix-the-thing
        ```
    * [ ] If there is no associated issue, [create one](https://github.com/DataBiosphere/toil/issues/new).
* [ ] Read through the code changes. Make sure that it doesn't have:
    * [ ] Addition of trailing whitespace.
    * [ ] New variable or member names in `camelCase` that want to be in `snake_case`.
    * [ ] New functions without [type hints](https://docs.python.org/3/library/typing.html).
    * [ ] New functions or classes without informative docstrings.
    * [ ] Changes to semantics not reflected in the relevant docstrings.
    * [ ] New or changed command line options for Toil workflows that are not reflected in `docs/running/{cliOptions,cwl,wdl}.rst` 
    * [ ] New features without tests.
* [ ] Comment on the lines of code where problems exist with a review comment. You can shift-click the line numbers in the diff to select multiple lines.
* [ ] Finish the review with an overall description of your opinion.

## Merger Checklist

* [ ] Make sure the PR passes tests.
* [ ] Make sure the PR has been reviewed **since its last modification**. If not, review it.
* [ ] Merge with the Github "Squash and merge" feature.
    * [ ] If there are multiple authors' commits, add [Co-authored-by](https://github.blog/2018-01-29-commit-together-with-co-authors/) to give credit to all contributing authors.
* [ ] Copy its recommended changelog entry to the [Draft Changelog](https://github.com/DataBiosphere/toil/wiki/Draft-Changelog).
* [ ] Append the issue number in parentheses to the changelog entry.

### CODE REVIEW

**Comment by mr-c** [comment]
@adamnovak I'd like to merge this PR, and then open a separate PR to add the final commit ID to `.git-blame-ignore-revs` which GitHub will then ignore automatically when doing a git blame view. The follow-on PR will also update `docs/contributing/contributing.rst` with the developer instructions to get the same benefit (`git config blame.ignoreRevsFile .git-blame-ignore-revs`).

**Comment by DailyDreaming** [comment]
@mr-c Oof, we can discuss this, but I'm not sure we're ready to convert.

Adding 18,000 fresh lines is quite verbose, and it doesn't look good to me.  Black would be better if users could configure it, since I don't like a lot of their decisions (including shorter line lengths and some arguably awkward wrapping).

**Review by adamnovak — CHANGES_REQUESTED** [review]
It looks like this ended up reverting some other semantically-important changes to the repo?

Speciffically e.g. https://github.com/DataBiosphere/toil/pull/4348/files#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52R44

**Comment by adamnovak** [comment]
Setting up a bypass for git blame for this is a good idea.

@DailyDreaming I think the *point* of Black is that users can't configure it; there shall be one true source code form for any Python abstract syntax tree, and the users shall never again need or be allowed to argue over it.

I'm still not sure that Black-ifying all the source is actually worth the trouble, though; we don't really waste a lot of time arguing over or fiddling with source code formatting, as far as I can tell. Most of our formatting-related costs go into thinking about whether we ought to add/adopt/enforce Black or not. (I've also spent a fairly substantial amount of time resolving merge conflicts when somebody rearranged imports and I added a new import.) Which to me says no, we shouldn't bother adding Black, it's not going to pay off.

But as long as we *haven't* adopted Black, the possibility of adopting it will remain, while if we *do* adopt it we'll almost certainly not decide to abandon it. So maybe the simplest possible solution is to do it, because if we don't there will be at least one more issue proposing that we do?

**Comment by adamnovak** [comment]
@mr-c You just ran the formatters once, right? Won't we need a process that somehow applies `black` to incoming changesets automatically, to prevent needing another giant Blackification PR in the future? How is that going to work?

I really want to avoid PRs failing *because* they didn't follow Black, because that makes e.g. the Github web editor kind of useless, and PRs from external contributors without our preferred dev setup much more of a pain to land. Maybe there's some kind of BlackBot we can turn lose on incoming PRs?

**Comment by mr-c** [comment]
> Black would be better if users could configure it, since I don't like a lot of their decisions (including shorter line lengths and some arguably awkward wrapping).

Line length is configurable with `--line-length 100` or via the following in `pyproject.toml`:
``` toml
[tool.black]
line-length = 100
```

Honestly, I don't care what tool is used, as long as it is comprehensive. `black` is used by almost all of the other projects I work on, so that would be a nice choice :-)

File: Makefile [hunk 1]
```diff
@@ -267,13 +266,10 @@ remove_unused_imports: $(PYSOURCES)
 remove_trailing_whitespace:
 	$(CURDIR)/contrib/admin/remove_trailing_whitespace.py
 
-format: $(wildcard src/toil/cwl/*.py)
+format: $(PYSOURCES)
```
Comment (mr-c) [inline Makefile:269]: @adamnovak Yes, here is the relevant line

**Comment by adamnovak** [comment]
I'm going to close this in favor of #4777 which changes the formatter only.

**Pull request closed without merging**